api/envoy/extensions/filters/http/grpc_field_extraction/v3/config.proto

```diff
@@ -0,0 +1,123 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.http.grpc_field_extraction.v3;
+
+import "envoy/config/core/v3/base.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.http.grpc_field_extraction.v3";
+option java_outer_classname = "ConfigProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/grpc_field_extraction/v3;grpc_field_extractionv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: gRPC Field Extraction]
+// gRPC Field Extraction :ref:`configuration overview
+// <config_http_filters_grpc_field_extraction>`.
+// [#extension: envoy.filters.http.grpc_field_extraction]
+
+// GrpcFieldExtraction filter supports extracting the fields from first gRPC
+// request message no matter if it is unary or streaming and it write the result
+// to the destination, for which currently only the static dynamic metadata `envoy.filters.http.grpc_field_extraction` is supported.
```

Comment:
Is it going to be desirable to support this filter in proxyless gRPC?  If so, note that gRPC does not support the same concept of metadata that Envoy does, so this approach won't work there.

Reply:
I feel no need to document that. As dynamic metadata is widely used within Envoy and I haven't seen other apis declaring it is not supported in proxyless gRPC. Also, this is not a hard blocker and we can easily change the filter to inject results into header if we want.